### PR TITLE
Add encryption checkbox to ransomware config page

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do? 

Addresses part of #1239  

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing? 
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running the Island (see screenshot below). Enabled/disabled encryption and ran ransomware to check it works.
* [x] If applicable, add screenshots or log transcripts of the feature working

image

### CODE REVIEW

File: monkey/infection_monkey/ransomware/ransomware_payload.py [hunk 1]
```diff
@@ -17,7 +17,10 @@
 
 class RansomewarePayload:
     def __init__(self, config: dict, telemetry_messenger: ITelemetryMessenger):
-        target_directories = config["directories"]
+        self.should_encrypt = config["encryption"]["should_encrypt"]
```
Comment (ilija-lazoroski) [inline monkey/infection_monkey/ransomware/ransomware_payload.py:20]: `should` is kind of confusing. Maybe `to_encrypt` or not, the question is now :sweat_smile:

File: monkey/tests/unit_tests/infection_monkey/ransomware/test_ransomware_payload.py [hunk 2]
```diff
@@ -141,6 +147,26 @@ def test_skip_already_encrypted_file(ransomware_target, ransomware_payload):
     )
 
 
+def test_encryption_skipped_if_configured_false(ransomware_target):
```
Comment (ilija-lazoroski) [inline monkey/tests/unit_tests/infection_monkey/ransomware/test_ransomware_payload.py:150]: Shouldn't we also have test for running the encryption using the checkbox ?

Comment (shreyamalviya) [reply monkey/tests/unit_tests/infection_monkey/ransomware/test_ransomware_payload.py:150]: All of the other tests use the `ransomware_payload_config` fixture which has `should_encrypt` as `True`, i.e. checkbox is checked.

File: monkey/infection_monkey/ransomware/ransomware_payload.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/infection_monkey/ransomware/ransomware_payload.py:20]: What about "enabled" ?

File: monkey/monkey_island/cc/services/config_schema/ransomware.py [hunk 3]
```diff
@@ -2,32 +2,45 @@
...
-                    "type": "string",
-                    "default": "",
-                    "description": "Files in the specified directory will be encrypted "
-                    "using bitflip to simulate ransomware.",
+                "should_encrypt": {
+                    "title": "Encrypt files",
+                    "type": "boolean",
+                    "default": True,
+                    "description": "Selected files will be encrypted using bitflip to simulate "
+                    "ransomware. Enter target directories below.",
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/config_schema/ransomware.py:14]: What about, "Ransomware encryption will be simulated by flipping every bit in the files contained within the target directories." ?

Maybe we should then change the below descriptions. I've made some suggestions.

File: monkey/monkey_island/cc/services/config_schema/ransomware.py [hunk 4]
```diff
@@ -2,32 +2,45 @@
...
-                    "type": "string",
-                    "default": "",
-                    "description": "Files in the specified directory will be encrypted "
-                    "using bitflip to simulate ransomware.",
+                "directories": {
+                    "title": "Directories to encrypt",
+                    "type": "object",
+                    "properties": {
+                        "linux_dir": {
+                            "title": "Linux encryptable directory",
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/config_schema/ransomware.py:21]:
```suggestion
                            "title": "Linux target directory",
```

File: monkey/monkey_island/cc/services/config_schema/ransomware.py [hunk 5]
```diff
@@ -2,32 +2,45 @@
...
+                            "type": "string",
+                            "default": "",
+                            "description": "Files in the specified directory will be encrypted "
+                            "using bitflip to simulate ransomware.",
+                        },
+                        "windows_dir": {
+                            "title": "Windows encryptable directory",
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/config_schema/ransomware.py:28]:
```suggestion
                            "title": "Windows target directory",
```

File: monkey/monkey_island/cc/services/config_schema/ransomware.py [hunk 6]
```diff
@@ -2,32 +2,45 @@
...
+                "directories": {
+                    "title": "Directories to encrypt",
+                    "type": "object",
+                    "properties": {
+                        "linux_dir": {
+                            "title": "Linux encryptable directory",
+                            "type": "string",
+                            "default": "",
+                            "description": "Files in the specified directory will be encrypted "
+                            "using bitflip to simulate ransomware.",
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/config_schema/ransomware.py:25]: "Ransomware encryption will be simulated by flipping every bit in the files contained within the specified directory"

File: monkey/monkey_island/cc/services/config_schema/ransomware.py [hunk 7]
```diff
@@ -2,32 +2,45 @@
...
+                        },
+                        "windows_dir": {
+                            "title": "Windows encryptable directory",
+                            "type": "string",
+                            "default": "",
+                            "description": "Files in the specified directory will be encrypted "
+                            "using bitflip to simulate ransomware.",
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/config_schema/ransomware.py:32]: "Ransomware encryption will be simulated by flipping every bit in the files contained within the specified directory"

File: monkey/tests/unit_tests/infection_monkey/ransomware/test_ransomware_payload.py (hunk 2 above)
Comment (mssalvatore) [reply monkey/tests/unit_tests/infection_monkey/ransomware/test_ransomware_payload.py:150]: We should have a test that verifies no encryption occurs if it's not enabled. See the `test_readme_*()` tests on the develop branch for examples.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/services/config_schema/ransomware.py (hunk 7 above)
Comment (shreyamalviya) [reply monkey/monkey_island/cc/services/config_schema/ransomware.py:32]: That seems repetitive to me. I modified it to: "Files in the specified directory will be encrypted using bitflip. If no directory is specified, no files will be encrypted."

**Review by ilija-lazoroski — APPROVED** [review]

File: monkey/monkey_island/cc/services/config_schema/ransomware.py (hunk 7 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/services/config_schema/ransomware.py]: I think "encrypted using bitflip" is awkwardly worded. "Bitflip" isn't an established encryption mechanism like AES.

Comment (mssalvatore) [reply monkey/monkey_island/cc/services/config_schema/ransomware.py]: I settled on, "A path to a directory on Windows systems that can be used to safely simulate the encryption behavior of ransomware. If no directory is specified, no files will be encrypted." We can always change it later.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**